### ISSUE TITLE
Twitter Report: „Es gibt Neuigkeiten von Ihrer Corona-Warn-App“ notification is sent repeatedly with red risk status

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#4211]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: moto g⁷ power (XT1955-4)
- Android version: 10
- App version: 2.11.2

## Describe the bug

This notification is send repeatedly (after clicking on it / dismissing it, it comes back after ca. 2h):

![image](https://user-images.githubusercontent.com/67682506/137072778-924ee279-8b4f-45b2-ba7f-929f32f44077.jpeg)

The user has an increased risk encounter from 02.10.2021, there the notification was send correctly and made the user aware of the encounter, however, now the notification is send without any reason again and again.

## Steps to reproduce the issue

1. Get a high risk encounter & let the CWA carry out a risk check in the background so that you get a push notification about the encounter.
2. Click on the notification 
3. See that the app opens and displays the encounter
4. Close the app
5. See that the notification comes back after 2h
6. Click on the notification
7. See that nothing has changed and the notification was send without any reason

## Expected behaviour

The notification about an increased risk encounter should only be send once.

## Additional context

This bug report is based on the feedback a user shared with me on Twitter: https://twitter.com/eillonnwy/status/1447637747523629060?s=21

---
Internal Tracking ID: [EXPOSUREAPP-10749](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10749)
Internal Tracking ID: [EXPOSUREAPP-12908](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12908)

**Comment by Ein-Tim** [comment]
@dsarkar 

Could we please reopen this issue? 
The bug does still appear, as recent reports from Twitter users show.

This leads to the misinterpretation that the users think that they had another risky encounter on the same day, as the day which is already shown to them. 

This should be investigated & fixed.

**Comment by Ein-Tim** [comment]
@dsarkar 
Will you mirror this issue to JIRA?

**Comment by Ein-Tim** [comment]
Another Twitter user experiencing the same issue: https://twitter.com/DasMuetze/status/1460230095403110403?s=20

@dsarkar Please, could we get this mirrored to JIRA?

**Comment by ein-christoph** [comment]
For four days in a row now I am getting the same notification at least twice a day. When I open the App either nothing has changed or it says "Exposure check failed". However I pretty sure the app had constant internet access and it is performing the check without any problems if I manually press "Restart" but even after the manually trigger check no new information or encounters show up (Currently having 3 encounters with high risk, one with low risk).

Please get this issue fixed because receiving up to five (!) notifications a day without an apparent reason is really annoying.

# Technical details

Device name: One Plus 7 Pro

Android version: 11

App version: 2.13.3

**Comment by Ein-Tim** [comment]
Another Twitter report: https://twitter.com/toadinthehole18/status/1462518004395167748?s=21

Can we get this mirrored to JIRA & investigated by the devs? Thanks @dsarkar!

**Comment by larswmh** [comment]
Hi @Ein-Tim @sodoku @ein-christoph, thanks for your reports. We have created an internal ticket and will raise this topic internally.
Internal Tracking ID: [EXPOSUREAPP-10749](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10749)

---
Corona-Warn-App Open Source Team

**Comment by vaubaehn** [comment]
Hi @larswmh , welcome and nice to have you onboard! Enjoy your ride with the lively CWA community.
Would you mind to set the label 'mirrored to jira' for this issue?
Thanks and best wishes
v.

**Comment by Ein-Tim** [comment]
@larswmh 

Btw, you could also make your membership of the CWA organization public, than you have the nice little "Member" badge next to your name when writing a comment (-;

This is always good for new/one time contributes, it helps them to differentiate between the community and the community managers (-:
But this is just a little point 🙂

**Comment by larswmh** [comment]
@vaubaehn thanks for welcoming me and reminding me to add the mirrored-to-jira label, it has now been added.

@Ein-Tim thanks for telling me, I have now changed my membership visibility to public.

**Comment by dsarkar** [comment]
@Ein-Tim Thanks for forwarding the report. Currently, I have no knowledge about this issue being reproducible. I will forward the case to the internal ticket.

**Comment by dsarkar** [comment]
FYI PR: https://github.com/corona-warn-app/cwa-app-android/pull/4597

**Comment by Ein-Tim** [comment]
A user on Twitter confirmed that version 2.16 fixed this: https://twitter.com/quereinsteiger/status/1484400969068785664?s=21

**Comment by dsarkar** [comment]
@Ein-Tim Thanks for the feedback. Forwarded to the internal ticket.

**Comment by NKarolak** [comment]
Unfortunately, the case is not solved yet.

Me and my boyfriend, we are on "status red" for over a week already.
This morning, our two (different) Android phones received the notification again. Clicking on it forwarded us to the main status page. It appears that the notification was NOT sent at the same time as the last contact check. Plus, according to the ENF, there was no new possible encounter detected today.

**Comment by dsarkar** [comment]
@NKarolak Thanks.

**Comment by NKarolak** [comment]
Correct. 

The status has been checked and updated regularly ("Updated": Today, ...), but the high risk encounter was still on the 19th February only. Also verified as per contact diary.

**Comment by dsarkar** [comment]
@NKarolak OK, thanks for your feedback.

**Comment by NKarolak** [comment]
Since today, I'm back on green again. I didn't receive any more false notification.

Btw there is a pattern. I had also red status back in December, whereas my boyfriend (we were on the same risk event) was on status green (low risk).
A week or so after the initial red risk check, I received the notification once again, whereas my boyfriend did not.

Hence I conclude: this must be somehow reproducible for people with red status since about 1 week.

**Comment by dsarkar** [comment]
@NKarolak Thanks for the info, will be forwarded. How often did you get the notification? While you had a red card, once per day? Same on both phones?

**Comment by NKarolak** [comment]
This time, I got the false extra notification only once, and while still in status red.
The very same for my boyfriend: same risk event, hence same red status since the same period, and we got our false notification almost at the same time on the same day. Just on different phones.

**Comment by ein-christoph** [comment]
I also received a "false" notification again today. I have been in status red for about 14 days now with two high-risk encounters (QR-Code checkins) one 14 days ago the other 13 days ago. Today I received the "Es gibt Neuigkeiten von der Corona-War-App" notification again without a new encounter showing up in the app.

Phone: Oneplus 7 Pro
Android-Version: 11
CWA-Version: 2.18.1

**Comment by Ein-Tim** [comment]
I can confirm this issue on my 
- Huawei P Smart 2019 with
- Android 10 and
- Version 2.18.1 of CWA installed

**Comment by Ein-Tim** [comment]
@dsarkar Please remove the `Fix 2.16` label. 

A Twitter user is also confirming this issue: https://twitter.com/pichunter_harry/status/1501882740689584135?s=20&t=_urY3RTX-Q7am25JHgkqPg

**Comment by MikeMcC399** [comment]
@Ein-Tim 
BTW "is send" in the issue title should be "is sent" ...

**Comment by vanitasvitae** [comment]
I'm having the same issue but on the "Corona Contact Tracing Germany"-fork.
Might this happen, if I had contact with more than one infected person? So that a new warning is issued for every contact, as soon as they decide to warn others using the app?

**Comment by Ein-Tim** [comment]
@dsarkar @larswmh 

Would you please inform the dev team that another fix is necessary because of a regression from the fix in 2.16?

This is an issue which is visible to everyone having an increased risk in their app & is easily reproducible, still it remains unfixed for some reason. 

cc @mtwalli & @mlenkeit

**Comment by dsarkar** [comment]
@Ein-Tim is https://twitter.com/DanielFreund79/status/1504467659705921543?s=20&t=3-uk3rzDkjfCGZBkHQh1Lw using 2.18, or at least 2.16?

**Comment by thomasaugsten** [comment]
I think it is a duplicate of 
https://github.com/corona-warn-app/cwa-app-android/issues/4925

**Comment by Ein-Tim** [comment]
@thomasaugsten No, here we have the issue that users are getting push notification for an increased risk status although they already received a push notification and already saw their increased risk status.

**Comment by thomasaugsten** [comment]
But if the risk switches in the background to green and red again he will receive a new push notification.
But we need a error report to confirm this

**Comment by Ein-Tim** [comment]
This has already been fixed in version 2.16 before so I think this is just a regression.

I will ask one of the users to activate error logging but it is not certain that they will receive another push notification. When do you want them to submit the log?

**Comment by thomasaugsten** [comment]
When they received a new push notification

**Comment by Ein-Tim** [comment]
@thomasaugsten 

One user provided an error report with the ID 739E1B411214D755ABEE.

I wonder if you can't reproduce this yourself, I think this affects all users.

Will you take a look at the log?

I asked the user for a timespan when the notification cam in.

cc @mlenkeit

**Comment by Ein-Tim** [comment]
@dsarkar All users are experiencing this, there are more than 3 reports on Twitter in the last 20 min. This should be easy to reproduce and to fix.

**Comment by Ein-Tim** [comment]
Twitter Reports (this list will be extended over time):

- https://twitter.com/Jahleesu/status/1504816040261394483?s=20&t=wjotoi3iNBwUk5y0W_CGjw
- https://twitter.com/Patjouk/status/1504798087629447186?s=20&t=wjotoi3iNBwUk5y0W_CGjw
- https://twitter.com/JaneraTiciano/status/1504786782595866625?s=20&t=wjotoi3iNBwUk5y0W_CGjw

**Comment by Ein-Tim** [comment]
> To me a confirmation that repeated notifications mean multiple contacts on the same day would be enough.

But that's actually not the case and the app is just buggy here.

**Comment by Ein-Tim** [comment]
@mlenkeit It seems like you are the correct person to ping here. Please, could you take a look at https://github.com/corona-warn-app/cwa-app-android/issues/4211#issuecomment-1072438473? 

Also see https://github.com/corona-warn-app/cwa-app-android/issues/4211#issuecomment-1072448151, this seems to be a very common bug which should be very easy to reproduce.

I honestly expect a fix for this either in 2.20 or better in 2.19. Everything else is not acceptable for such an issue.

**Comment by mlenkeit** [comment]
We're looking into it already. It indeed doesn't look like it's related to https://github.com/corona-warn-app/cwa-app-android/issues/4925.

**Comment by Ein-Tim** [comment]
Thanks Max! Sorry for making such a noise on a Friday afternoon! Hope you have a relaxed week-end at least! :D

**Comment by mtwalli** [comment]
@Ein-Tim  Unfortunately the logs from https://github.com/corona-warn-app/cwa-app-android/issues/4211#issuecomment-1072438473 is not telling that there is an issue regrarding that matter. I see `24` logs of risk level change all of them say no change ,but only one which was changed from LOW to HIGH and that led one notification was triggered as expected.
```log
2022-03-18T11:09:40.156Z D/CombinedRiskLevelChangeDetector: Risk changed=true from=LOW_RISK to=INCREASED_RISK
2022-03-18T11:09:40.160Z D/CombinedRiskLevelChangeDetector: Notification Permission = true
```
if are in contact with the users, logs that covers longer period of risk calculations would be extremely appreciated

**Comment by Ein-Tim** [comment]
@mtwalli 

Thank you for getting back! Unfortunately the user do eon to have a longer log.
But as you can see in https://github.com/corona-warn-app/cwa-app-android/issues/4211#issuecomment-1072448151, this seems to affect all users. I highly recommend to attempt reproducing it by yourself and/or checking if the PR which fixed this in 2.16 (https://github.com/corona-warn-app/cwa-app-android/pull/4597) was somehow reverted, etc..

**Comment by MikeMcC399** [comment]
@Ein-Tim 
> Unfortunately the user do eon to have a longer log. 

There seems to be a typo in that sentence, so I don't know what it means.

> But as you can see in [#4211 (comment)](https://github.com/corona-warn-app/cwa-app-android/issues/4211#issuecomment-1072448151), this seems to affect all users.

I have currently a red risk tile showing with "Exposures on 5 days with increased risks" and I am not being hit by repeated notifications. This is on Android 11, CWA 2.19.0 and ENS Version 1.8 (220618000). So although you may have multiple reports of the issue, it doesn't affect **all** users, otherwise I would be having the problem as well.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

Thank you! I fixed the typo. 

>  have currently a red risk tile showing with "Exposures on 5 days with increased risks" and I am not being hit by repeated notifications. This is on Android 11, CWA 2.19.0 and ENS Version 1.8 (220618000). So although you may have multiple reports of the issue, it doesn't affect all users, otherwise I would be having the problem as well.

Did you receive the notification for these encounters on a version pre 2.19?

**Comment by MikeMcC399** [comment]
@Ein-Tim 

Did you receive the notification for these encounters on a version pre 2.19?

- the notification(s) for 4 risk days were received using 2.18.1
- 1 notification for a new 5th risk day was received using 2.19.0

**Comment by Ein-Tim** [comment]
>> Did you receive the notification for these encounters on a version pre 2.19?

> the notification(s) for 4 risk days were received using 2.18.1
1 notification for a new 5th risk day was received using 2.19.0

You should not receive any notifications if your risk status is already increased. Thus you are actually also affected by this bug.

**Comment by MikeMcC399** [comment]
> You should not receive any notifications for a changed risk status if your risk status is already increased. Thus you are actually also affected by this bug. Or did your app went back to green between the risk encounters?

It seems you are saying that I should not get a new notification if a new risk day is added. If I have a new risk encounter on a different day, I would like to be notified about it, even if my risk status was already red. I would then re-evaluate my risk based on what I was doing that day and consider testing / re-testing / quarantine.

My risk status did not go back to green, no.

According to the original description of this issue I understood that the problem was repeated notifications even though no new risk days had been added. Perhaps I misinterpreted the issue?

**Comment by Ein-Tim** [comment]
@MikeMcC399 

> I will pay attention for the future, but for the history I think I have to drop out of making statements.

Okay, thank you! Maybe you would also like to activate error logging if there is enough free space available on your phone? 

Have a nice evening!

**Comment by Ein-Tim** [comment]
This Twitter user [says](https://twitter.com/TimoBeil_/status/1505919869400801289?s=20&t=RMc3Y2v7sJ-d5jyBjfC90w):

> Durchgehend rot! Immer wieder dieselbe Meldung Neuigkeiten zur App. Ich musste die Benachrichtigung abschalten, weil es mir auf den Sack geht nach 10-20 Mal! Behebt den Fehler!

I wonder if this is really not reproducible on any of the devices from the developers @thomasaugsten, @mlenkeit or @mtwalli. 

I asked him for an error log and am in contact with him. I will keep you posted.

**Comment by Ein-Tim** [comment]
The Twitter user [@TimoBeil_](https://twitter.com/TimoBeil_) now provided two logs, both were send at around the same time and should have captured how the app sends two wrong notifications.

Here are the IDs: 

ID 1: 487999CEF63C33DBB4D0

ID 2: E00332B70126AF5FBC99

**Comment by thomasaugsten** [comment]
Die ID: E00332B70126AF5FBC99 existiert nicht wahrscheinlich fehlt ein Zeichen

**Comment by Ein-Tim** [comment]
@thomasaugsten 

Sorry, mein Fehler, da war ein Typo drin. Das ist die richtige: E00332B7D126AF5FBC99

**Comment by mtwalli** [comment]
@Ein-Tim  last report does not show any information about the risk level change. Answering your questions above , it is difficult to produce this issue on a testing device for two main reasons:
1- We develop different features all the time for different app versions, so we install /uninstall the app for short periods
2- Environment for testing device is not ideal for this case, there should be encounters which is not happening at Home 

Pr (https://github.com/corona-warn-app/cwa-app-android/pull/4597) fix is in place , which was fixing the case of risk failure and then risk became High or Low also if the same risk reported repeatedly

**Comment by Ein-Tim** [comment]
> last report does not show any information about the risk level change

Is there anything else which could explain why the user saw two messages during the log was running?

**Comment by MikeMcC399** [comment]
@Ein-Tim

Do you think that it is the same notification which is reappearing after approximately 2 hours, even though it has been dismissed or do you think that the app is creating a new notification and displaying it?

**Comment by MikeMcC399** [comment]
I'm wondering also why a notification would be repeated after 2 hours. If it was due to a new risk calculations, then that should not happen for 4 hours on WiFi.

**Comment by Ein-Tim** [comment]
@mlenkeit 

As promised, I have many users on Twitter and I'm getting error reports from many of them. Here's another one from [this](https://twitter.com/juliac_dehemm/status/1506531301830111237?s=21) user: 

ID: 5527147EF11BF4AF3068

Could you explain why a bug this easy to reproduce is not fixed. I imagine that you need the following to reproduce this:
1. A device with an install of version 2.19.1 of CWA
2. An increased risk encounter (I'm sure you have the ability to fake one)
3. A logger which logs what the app is doing and why

Or am I underestimating how hard it is to reproduce this?

Could you at least let us know if you are able to reproduce it and if someone is already working on a fix?

**Comment by Ein-Tim** [comment]
@thomasaugsten 

Okay, thank you!

**Comment by MikeMcC399** [comment]
I got a notification today March 24, 2022 at 10:33 with a red dot on the status icon. The red risk tile says I have "Increased Risk". It was showing this status previously. The contact was "Most recently on 18 Mar 2022", which is also not a change. The last exposure check was 07:33, which is exactly three hours before the new notification. The Google exposure check log (Check details) shows Number of keys 15937 and Number of matches 0 for the exposure check with Timestamp 24 March 2022, 07:34.

Does the exact three hour difference between last exposure check and the notification give any clue?

CWA would have been running in the foreground because I had recently opened it to view the statistic tiles.

I am going to enable error reports on this device now in case the issue reappears.

For anybody else with error reports enabled I would think it would be quite important to make a note of the exact time an unexpected notification appears, so that it can be traced in the log.  (The log uses UTC timestamps - currently 1 hour earlier than local CET time - so pay regard to this when matching events together with log entries.)

**Comment by thomasaugsten** [comment]
The latest error reports are not useful because there are only 1 minute long. We need an error report which runs over 8h.

**Comment by Ein-Tim** [comment]
@thomasaugsten Okay, I thought they just need to activate logging and wait until they receive a notification

**Comment by thomasaugsten** [comment]
We need minimum two notifications

**Comment by MikeMcC399** [comment]
> We need minimum two notifications

That could take a while in my case, since the previous notification before today was 6 days ago.

**Comment by Ein-Tim** [comment]
Okay I will ask one user again.

**Comment by vaubaehn** [comment]
Hi @MikeMcC399 ,
could you please cross-check these two informations from your past comments:

> The red risk tile says I have "Increased Risk". It was showing this status previously. The contact was "Most recently on 18 Mar 2022", which is also not a change.

> That could take a while in my case, since the previous notification before today was 6 days ago.

That would mean that your notification 6 days ago on 18 Mar 2022 showed an increased risk contact for 18 Mar 2022?
It is theoretically possible, when your device recorded the contact with that person for that day, and when that person shared the positive test result on that same day. Due to Same-Day-TEKs that is theoretically possible but unlikely, however.
I'm asking to definitely rule out that there was actually a change in the date of the risk contact that might have been overlooked.

But in any case it's interesting that the notification came 3 hours after the last matching of Diagnosis Keys.

**Comment by MikeMcC399** [comment]
@vaubaehn 
I should probably have said that the notification was 5 days ago. One day after the match. I did not make a written note of the notification, so I no longer have any way of verifying. To do any sort of troubleshooting with recorded data I am going to have to wait for repeat notifications, if they even occur, and then make sure I make a written note of when they happen.

**Comment by vaubaehn** [comment]
@thomasaugsten - Just asking into the blue:
We discussed before, that Exposure Windows expire on an hourly basis.
Would following scenario be possible?
User records an unknown number of contacts for the same day x-6d.
At day x-5d, some of these contacts from day x-6d share their Diagnosis Keys from infections where symptoms started somewhat earlier so that they have a low transmission risk level. But the addition of all these Exposure Windows lead to an increased risk, just matching with the cut-off for increased risk.
At day x-4d, one single risk contact from day x-6d shares its Diagnosis Keys, and transmission risk level results in an increased risk. The risk tile shouldn't change at all: it stays red, and number of days stays '1'.
At day x+3hrs, one (or more) of the low-risk contacts from day x-6d that shared their keys on day x-5d expires. The Exposure Windows of the remaining contacts that shared keys on x-5d now result in a low risk.
But the risk tile wouldn't change again, as there is still the increased risk contaxt from the person that shared keys on day x-4d.
Could there be a bug in risk calculation/notification that "perceives" that change as a need to trigger a new notification, i.e. that single increased risk contact now "gets a higher priority" and causes the notification?

**Comment by mtwalli** [comment]
@Ein-Tim , I got an error report that shows multiple notifications, but not for risk changes. it shows multiple notification for booster rule. I think those users are also affected by https://github.com/corona-warn-app/cwa-app-android/issues/4959. 
That explains why some user are reporting this issue again with 2.19.0 when was released. 
You can confirm with them:
- If they have some certificates will slightly different names
- Notification is triggered when they bring the app to foreground [from background calculation should happen every 24 hours]

if that is the case , then please advice them to upgrade to 2.19.2 (Hopefully the issue is gone🤞)

**Comment by Ein-Tim** [comment]
@mtwalli 

Thank you!

I can confirm both things and asked the user to confirm that version 2.19.2 fixed this issue.

**Comment by Ein-Tim** [comment]
User confirmed, version 2.19.2 fixed this issue.

**Comment by mtwalli** [comment]
Good 👍🏼, are all reports have the same cause? or some users are `really` experiencing risk change notifications?

**Comment by Ein-Tim** [comment]
I will ask them whether 2.19.2 fixed the issue.

**Comment by Ein-Tim** [comment]
@NKarolak Are you still experiencing this issue? Same question to @MikeMcC399.

**Comment by NKarolak** [comment]
> @NKarolak Are you still experiencing this issue?
Cannot tell. I was only affected while in status red. And I'm currently not.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

I still have a red tile and I have had no more notifications since Mar 24, 2022. I am running CWA 2.19.2.

**Comment by Ein-Tim** [comment]
Thank you @MikeMcC399 & @NKarolak!

All Twitter users I asked and responded also confirmed that the bug is resolved. However, one user said he still received a faulty notification: https://twitter.com/Wombelbombel/status/1508152790845116418?s=20&t=jIa9I1otixJugIF54ybKgA

FYI @mtwalli

**Comment by mtwalli** [comment]
@Ein-Tim Thank you, could you please ask the user for an 8 hours error report? I can't really say if he is facing different cause.

**Comment by Ein-Tim** [comment]
@mtwalli I asked but no feedback yet...

**Comment by MikeMcC399** [comment]
@mtwalli 

On CWA Android 2.19.2 I received a notification of status change today at 15:44 CEST. Both before and after this notification my status was red and the last day (On 18 Mar 2022) has not changed.

I have had error reporting running on this device for one week since Mar 24, 2022. I activated the error report after the previous notification on Mar 24, 2022. Whilst the error report has been gathering logs there has been only one notification and that was today, Mar 31, 2022 at 15:44 CEST.

I found the following in the log file: (13:44Z (UTC) = 15:44 CEST).

```
2022-03-31T13:44:07.094Z D/RiskLevelTask: Risklevel calculated: EwAggregatedRiskResult(totalRiskLevel=HIGH, totalMinimumDistinctEncountersWithLowRisk=1, totalMinimumDistinctEncountersWithHighRisk=1, mostRecentDateWithLowRisk=null, mostRecentDateWithHighRisk=2022-03-18T00:00:00.000Z, numberOfDaysWithLowRisk=0, numberOfDaysWithHighRisk=1, exposureWindowDayRisks=[ExposureWindowDayRisk(dateMillisSinceEpoch=1647561600000, riskLevel=HIGH, minimumDistinctEncountersWithLowRisk=1, minimumDistinctEncountersWithHighRisk=1)])
2022-03-31T13:44:07.094Z I/RiskLevelTask: Risk is increased!
2022-03-31T13:44:07.106Z I/RiskLevelTask: Risk level determined: RiskLevelTaskResult(calculatedAt=2022-03-31T13:44:07.094Z, failureReason=null, ewAggregatedRiskResult=EwAggregatedRiskResult(totalRiskLevel=HIGH, totalMinimumDistinctEncountersWithLowRisk=1, totalMinimumDistinctEncountersWithHighRisk=1, mostRecentDateWithLowRisk=null, mostRecentDateWithHighRisk=2022-03-18T00:00:00.000Z, numberOfDaysWithLowRisk=0, numberOfDaysWithHighRisk=1, exposureWindowDayRisks=[ExposureWindowDayRisk(dateMillisSinceEpoch=1647561600000, riskLevel=HIGH, minimumDistinctEncountersWithLowRisk=1, minimumDistinctEncountersWithHighRisk=1)]), exposureWindows.size=60)
 ```

Two hours previous to that, the `RiskLevelTask` logs the following and there was no notification:

```
2022-03-31T11:42:52.621Z D/RiskLevelTask: Risklevel calculated: EwAggregatedRiskResult(totalRiskLevel=HIGH, totalMinimumDistinctEncountersWithLowRisk=1, totalMinimumDistinctEncountersWithHighRisk=1, mostRecentDateWithLowRisk=null, mostRecentDateWithHighRisk=2022-03-18T00:00:00.000Z, numberOfDaysWithLowRisk=0, numberOfDaysWithHighRisk=1, exposureWindowDayRisks=[ExposureWindowDayRisk(dateMillisSinceEpoch=1647561600000, riskLevel=HIGH, minimumDistinctEncountersWithLowRisk=1, minimumDistinctEncountersWithHighRisk=1)])
2022-03-31T11:42:52.622Z I/RiskLevelTask: Risk is increased!
2022-03-31T11:42:52.623Z I/RiskLevelTask: Risk level determined: RiskLevelTaskResult(calculatedAt=2022-03-31T11:42:52.623Z, failureReason=null, ewAggregatedRiskResult=EwAggregatedRiskResult(totalRiskLevel=HIGH, totalMinimumDistinctEncountersWithLowRisk=1, totalMinimumDistinctEncountersWithHighRisk=1, mostRecentDateWithLowRisk=null, mostRecentDateWithHighRisk=2022-03-18T00:00:00.000Z, numberOfDaysWithLowRisk=0, numberOfDaysWithHighRisk=1, exposureWindowDayRisks=[ExposureWindowDayRisk(dateMillisSinceEpoch=1647561600000, riskLevel=HIGH, minimumDistinctEncountersWithLowRisk=1, minimumDistinctEncountersWithHighRisk=1)]), exposureWindows.size=60)
```

Would you like me to upload the error report and provide the ID here?

**Comment by MikeMcC399** [comment]
@mtwalli 
I received another notification today with effectively no change in the risk status. I still have error report enabled. If by any chance you would like an updated error report covering about 8 days, please let me know, but I guess you probably have everything you need right now.

**Comment by mtwalli** [comment]
Yes, please do? I started looking at the first report. I see once that the risk was low and then became high, but I think you never saw the low risk state , right?

**Comment by MikeMcC399** [comment]
@mtwalli 
> Yes, please do? I started looking at the first report. I see once that the risk was low and then became high, but I think you never saw the low risk state , right?

The new ID is 964FA40BD3E1EEAE04E4. I hope that is helpful to you. The most recent notification today was at 10:14 Apr 1, 2022 (CEST) or 08:14 UTC.

That is correct. I only started the error report when the risk tile was already red. During this time the risk tile has always been red including right now. It was never green during the time period covered by the error report.

**Comment by svengabr** [comment]
App store reviews state the issue is not fixed with 2.19.2

> Zur Zeit bekomme ich ständig auf meinem Handy die Mitteilung, dass es Neuigkeiten von meiner Corona-Warn-App gäbe. Wenn ich dann draufklicke, hat sich aber nichts geändert und ich kann auch nichts erkennen. Was ist das?
by [Stein Bron](https://www.google.com/search?q=%22Stein+Bron%22) for v2.19.2 (2190201) Android 12.0, Galaxy s20 5g (x1s) via Google Play (DE) • April 1

> Bisher war ich zufrieden mit der App. Aber warum kriege ich gerade täglich die Meldung, es gäbe Neuigkeiten, aber dann gibt's keine Änderung bei Risikostatus?
by [Noo Name](https://www.google.com/search?q=%22Noo+Name%22) for v2.19.2 (2190201) Android 11.0, Galaxy a20e (a20e) via Google Play (DE) • April 1

> Es gibt Neuigkeiten...Es gibt Neuigkeiten...Es gibt Neuigkeiten...Es gibt Neuigkeiten... Was bitte soll denn diese Benachrichtigung??? Welche Neuigkeiten wo denn was denn was will diese app von mir??? In der app gibt es nichts neues also was soll das denn???
by [BrotboxGaming](https://www.google.com/search?q=%22BrotboxGaming%22) for v2.19.2 (2190201) Android 12.0, Galaxy s21 fe 5g (r9q) via Google Play (DE) • April 1

> Schlecht. Jeden Tag bekomme ich die Benachrichtigung " Es gibt Neuigkeiten von ihrer Warnapp" Ja die da bitte wären? Seit über 2 Wochen wird es mir Rot angezeigt. Solangsam könnte es mal weg gehen. Die tägliche Benachrichtigung ohne Grund nervt auch nur noch .
by [Celine v.](https://www.google.com/search?q=%22Celine+v.%22) for v2.19.2 (2190201) Android 12.0, Galaxy s21 fe 5g (r9q) via Google Play (DE) • April 1

**Comment by MikeMcC399** [comment]
Also in Twitter, which @Ein-Tim closely follows.

**Comment by Ein-Tim** [comment]
As @MikeMcC399 says, there is a spike of reports on Twitter at the moment. Not sure why, I'm getting in touch with these users, requesting information on their app & OS version as well as an error report.

**Comment by Ein-Tim** [comment]
FYI: Links to the Twitter reports I'm dealing with at the moment pasted into https://github.com/corona-warn-app/cwa-app-android/issues/4211#issuecomment-1072448151.

**Comment by Ein-Tim** [comment]
[This](https://twitter.com/BlackAlien/status/1509584381442244612?s=20&t=bBKk6EVuhgzVkBOg-_8dcA) Twitter user provided an Error report running for some time and capturing a worn notification which was send today before 9:30h. 


The report has the ID: 93571345008444A49A8D

**Comment by MikeMcC399** [comment]
There seems to be a lot of evidence that this is not an isolated issue, so hopefully the root cause can soon be found and fixed.

**Comment by uschindler** [comment]
I don't get the notification every two hours, but since two weeks regularly (approximately once per day). After clicking notification you get same encounter stats, also the diary does not change. At moment I have 2 encounters 1.5 weeks ago.

**Comment by MikeMcC399** [comment]
@mtwalli 
I got a final notification this morning at 10:10 Apr 2, 2022 informing me that my status had changed to low (green) due to the last exposure being more than 14 days ago (this is CWA 2.19.2). So I don't expect to provide any more data for the issue of spurious notifications in the near future.

It surprised me however that the status change didn't happen until 10:10. Why didn't the risk change happen at the start of today? There were two previous risk checks today: 01:50 and 06:06, so I would have expected the displayed risk to have changed at 06:06 at the latest, however that was not the case. Up until 10:10 the app was still displaying red, even though the exposure was on Mar 18, 2020 and could no longer be displayed in the Journal because the Journal only covers Mar 19 to Apr 2, 2022 inclusive and the date of exposure on Mar 18 is earlier than the first day displayed in the Journal.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

IIRC somewhere it was stated that the RPIs are deleted exactly 15x24h after they were received.
So that's why the encounter does not disappear at the start of the day but only when the RPIs, which caused the increased risk warning, are deleted.

**Comment by MikeMcC399** [comment]
@Ein-Tim 
> IIRC somewhere it was stated that the RPIs are deleted exactly 15x24h after they were received.
So that's why the encounter does not disappear at the start of the day but only when the RPIs, which caused the increased risk warning, are deleted.

Thank you! Since your comment got a 👍🏻 from @thomasaugsten I take it that this is the expected behaviour and I will not pursue the issue further.  I think this is a little inconsistent from a UI perspective however, since the red tile is telling you to look in the contact journal and there is no longer an exposure listed there.

**Comment by MikeMcC399** [comment]
@mtwalli 

It has been a couple of weeks since I supplied the error reports you asked for in https://github.com/corona-warn-app/cwa-app-android/issues/4211#issuecomment-1085637799. I'm currently not experiencing the issue, since I don't have a red tile showing. Nevertheless I wonder if you were able to spot any root cause to prevent the issue occurring in future?

**Comment by mlenkeit** [comment]
@MikeMcC399 unfortunately, we weren't able to pin-point the root cause yet, despite all the error reports.

**Comment by Ein-Tim** [comment]
I think it is especially strange that it seems to always occur in the wild but not during the tests the developers performed!

**Comment by mlenkeit** [comment]
> I think it is especially strange that it seems to always occur in the wild but not during the tests the developers performed!

@Ein-Tim ... there's also a whole testing squad, where this issue never showed up, despite a large variety of (Android) testing devices. It's frustrating to see such kind of errors in production and not being able to reproduce them.

**Comment by vaubaehn** [comment]
I now also experienced this issue.
I have a red tile for 27th of April, showing up first on 30th of April. (I was a

**Comment by vaubaehn** [comment]
I now also experienced this issue.
I have a red tile for 27th of April, showing up first on 30th of April. (I was at DMEA, where only around 10-20% of visitors and exhibitors wore masks... Has any of you SAP devs also been attending there?)
I'm on CCTG 2.14.x, and I was following the increasing numbers of risk contacts reported by microG (risk contacts with contribution to calculated risk level as well without contribution are displayed), calculated from subsequent new shared TEKs between Saturday and today.
From my observation, no new risk contacts have been added or subtracted since last night for the whole last 14 days, so there was no new risk level calculated. However, the notification showed up around 3 hours after the last download and provision of DKs to ENF.
So from my perspective looks like expiring or new TEKs and a risk level recalculation could be ruled out as the cause for the issue.

**Comment by dsarkar** [comment]
@vaubaehn Thanks. Forwarded to Internal Tracking ID: [EXPOSUREAPP-12908](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12908)

**Comment by vaubaehn** [comment]
@mtwalli (and @mlenkeit)
Is the value that triggers the notification emitted in a `Flow`?

**Comment by smares** [comment]
I just experienced this issue again today with 2.21.1 on Android 12, Pixel 6 Pro. I have a red tile since Friday because a colleague in my office has COVID. Today at around 9, I received a push notification that my CWA has news. Opened the app, but nothing new, not even in the journal. Last refresh was at around 7 in the morning. Just checked again, last update at 11:52, but still nothing new.

**Comment by dsarkar** [comment]
@smares Thanks, info was forwarded.

**Comment by mlenkeit** [comment]
FYI, we are observing a similar behavior in our test environment at the moment: the app configuration that the server provides appears to the client as _changed_ (although it didn't). In such cases where the app configuration changed, the risk encounters are newly evaluated and new notifications are issued.

However, while the the app configuration in our test environment changes at an unusual frequency (something that we are investigating), we do not observe this in the productive environment.

Long story short: we're not still sure of this is the root cause for the repeated notifications.

**Comment by MikeMcC399** [comment]
@dsarkar 

Since there are on-going reports of this issue, probably the https://github.com/corona-warn-app/cwa-app-android/labels/Fix%202%2E16 label should be removed.

**Comment by dsarkar** [comment]
FYI - a backend fix was deployed two days ago, which might fix this issue.

**Comment by Ein-Tim** [comment]
@dsarkar Do you have a link to the corresponding PR?

**Comment by svengabr** [comment]
I have just checked the linked Jira issue.

Jira Ticket is flagged as:
Resolution: `Fixed`
Status: `Completed`
Fix Version/s: `2.22`

Developer comment:
> ... backend fix was deployed today

> ... does https://github.com/corona-warn-app/cwa-server/pull/1821 include the fix?

> ... we could reproduce such a behavior and the PR you mentioned in the backend fixed the issue for us. The assumption is that it's now also resolved for the reported cases.

**Comment by Ein-Tim** [comment]
@mlenkeit, @svengabr, @thomasaugsten 

FYI: There are user reports on Twitter reporting that this issue occurs again on version 2.24. I'm trying to get more information and will open a new issue.

Twitter links:
- https://twitter.com/Fem_Lea/status/1543545152534450181?s=20&t=HEz5FuU3js66jxBVhxOPww
- https://twitter.com/Ich_halt_/status/1543567371742453765?s=20&t=Txynlpp7OCopueNuDSxfrw

**Comment by dsarkar** [comment]
@Ein-Tim Thanks for letting us know!

**Comment by dsarkar** [comment]
@Ein-Tim Did you get more info from these users
- they have a red or green warning
- how many certificates
- what kind of certificates they have in their app
- red notification dot on the tab and if so, on which one?

Thanks!

**Comment by Ein-Tim** [comment]
@dsarkar 

- No red or green warning
- I will ask the other questions
- Maybe one of the users will provide an error report.

**Comment by dsarkar** [comment]
@Ein-Tim 
- Indeed, an error report might be helpful if it records the notification
- Also it is relevant if these users have expired certificates

Thanks again for having a look at Twitter!

**Comment by Ein-Tim** [comment]
@dsarkar I have another Twitter user ([Link to Twitter](https://twitter.com/tantegreta1/status/1544256046235111426?s=20&t=9E8nF6YBqLPnrvpwPpPpGg)) with this problem. Here's a screenshot of the apps notifications. This does not seem normal. 

<img src="https://user-images.githubusercontent.com/67682506/177343426-d294a73b-8b33-43fa-8eaf-6f8c714eefd7.jpeg" width="300">

I asked him all questions you posted above.

**Comment by dsarkar** [comment]
@Ein-Tim Thanks for all the info and effort!
